source/FreeRTOS_IP.c

```diff
@@ -1772,31 +1772,32 @@ static eFrameProcessingResult_t prvProcessIPPacket( const IPPacket_t * pxIPPacke
              * entry. */
             if( ucProtocol != ( uint8_t ) ipPROTOCOL_UDP )
             {
-                /* Refresh the ARP cache with the IP/MAC-address of the received
-                 *  packet.  For UDP packets, this will be done later in
-                 *  xProcessReceivedUDPPacket(), as soon as it's know that the message
-                 *  will be handled.  This will prevent the ARP cache getting
-                 *  overwritten with the IP address of useless broadcast packets. */
-                if( pxIPPacket->xEthernetHeader.usFrameType == ipIPv6_FRAME_TYPE )
-                {
-                    vNDRefreshCacheEntry( &( pxIPPacket->xEthernetHeader.xSourceAddress ), &( pxIPHeader_IPv6->xSourceAddress ), pxNetworkBuffer->pxEndPoint );
-                }
-                else
-
                 if( xCheckRequiresARPResolution( pxNetworkBuffer ) == pdTRUE )
```

Comment:
Why is ARP resolution needed here for IPv6?

Reply:
Yes maybe we should give that function a different name like: `xCheckRequiresAddressResolution()`. In this PR I added code to handle both IPv4 and IPv6 packets.